pkg/volume/fc/fc_util.go

```diff
@@ -373,6 +380,15 @@ func (util *fcUtil) DetachBlockFCDisk(c fcDiskUnmapper, mapPath, devicePath stri
 	return nil
 }
 
+func (util *fcUtil) deleteMultipathDevice(exec utilexec.Interface, dmDevice string) error {
+	out, err := exec.Command("multipath", "-f", dmDevice).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to flush multipath device %s: %s\n%s", dmDevice, err, string(out))
```

Comment:
just want to confirm whether in certain cases, it should continue instead of return error (e.g., the error means device no longer exist?)

Reply:
Good catch! That's why iscsi volume plugin ignores the error code...

Reply:
If the device is missing:

* Filesystem volumes: operation_generator.go will stop with "The path isn't device path or doesn't exist. Skip checking device path: /dev/dm-0" and it won't call volume plugin at all. Therefore, if the multipath device is removed, but the individual paths are still there, nothing will remove them.

* Block volumes: if the device is missing, the next `DetachBlockFCDisk` will stop in `checkPathExists`, if not earlier in `operation_generator.go` (did not test it).
https://github.com/kubernetes/kubernetes/blob/6f8bdb7cc1e28585edb5d47212ffb4d0ab3bf546/pkg/volume/fc/fc_util.go#L301

iscsi volume plugin actually ignores the error code from "multipath -f", continuing deleting the device paths. Which may result in volume corruption, if the multipath device still exists.. Now is the question what is actually correct - try to remove the individual paths to fully detach the devices and risk corruption OR do not remove the paths at all? The paths should just sit in /dev, not used by anything.

My gut feeling is to copy iscsi approach and ignore the exit code.

Reply:
according to manpage
`-f     Flush (remove) a multipath device map specified as parameter, if unused.`

So if `multipath -f` fails, there is a good chance the device is still being used and removing it could cause data corruption. In this sense, the implementation in iscsi is indeed risky.